test/integration/emulator/TimeTurner.cpp

```diff
@@ -0,0 +1,143 @@
+#include "TimeTurner.h"
+#include "logger/Logger.h"
+#include <thread>
+
+namespace emulator
+{
+
+TimeTurner::TimeTurner() : _timestamp(100), _startTime(std::chrono::system_clock::now()), _running(true) {}
+
+void TimeTurner::nanoSleep(const uint64_t nanoSeconds)
+{
+    if (!_running)
+    {
+        std::this_thread::yield();
```

Comment:
We can let it run but we probably have threads at the end of a test. The thread did expect to pasue anyway so if there is nothing else to do it will continue.